src/utils/simplifier.js

```diff
@@ -103,14 +103,50 @@ function simplify(realm, value: Value, isCondition: boolean = false): Value {
         // (x: boolean) || false <=> x
         if (!y.mightNotBeFalse()) return op === "||" ? x : realm.intrinsics.false;
       }
-      if (
-        op === "||" &&
-        y instanceof AbstractValue &&
-        y.kind === "||" &&
-        x.equals(y.args[0]) &&
-        !y.args[1].mightNotBeTrue()
-      )
-        return y;
+      if (op === "||") {
+        if (y instanceof AbstractValue && y.kind === "||") {
+          // x || x || y <=> x || y
+          if (x.equals(y.args[0])) return y;
+          if (x instanceof AbstractValue && x.kind === "!") {
+            // !x0 || y0 || x0 <=> y0
```

Comment:
This seems wrong to me. `!x0 || ... || x0` is `true`, not equivalent to some random value.

Reply:
Sadly it is not even true. If x0 is truthy, the result is y0 || x0. If not, the result is true. The entire rule has to go.

Reply:
Correct. Moreover, only if isCondition is true.